apps/dot_ext/models.py

```diff
@@ -257,26 +257,38 @@ def save(self, *args, **kwargs):
                                 "application_name": self.name,
                                 "data_access_type_old": app_from_db.data_access_type,
                                 "data_access_type_new": self.data_access_type,
-                                "grant_delete_start": datetime.now().strftime("%m/%d/%Y, %H:%M:%S"),
+                                "grant_start": datetime.now().strftime("%m/%d/%Y, %H:%M:%S"),
                             })
                             logger.info(log_dict)
-                            dag_deleted = DataAccessGrant.objects.filter(application=self).delete()
+                            if self.has_one_time_only_data_access():
+                                dag_deleted = DataAccessGrant.objects.filter(application=self).delete()
+                                end_time = time.time()
+                                delete_stats = {
+                                    "elapsed_seconds": end_time - start_time,
+                                    "number_of_grant_deleted": dag_deleted[0],
+                                    "grant_delete_complete": datetime.now().strftime("%m/%d/%Y, %H:%M:%S")
+                                }
+                                log_dict.update(delete_stats)
+                            elif "THIRTEEN_MONTH" in self.data_access_type:
+                                grants = DataAccessGrant.objects.filter(application=self)
+                                for grant in grants:
+                                    grant.update_expiration_date(True)
```

Comment:
Not sure whether or not this boolean would be ignored for the param you just removed?
Otherwise, the rest looks good to me!

Reply:
@ajshred Getting this error when changing type to 13-month in the Django admin and saving:

```suggestion
                                    grant.update_expiration_date()
```

TypeError
--
  File "/code/apps/dot_ext/forms.py", line 188, in save
    app.save()
update_expiration_date() takes 1 positional argument but 2 were given
/code/apps/dot_ext/models.py, line 275, in save